### ISSUE TITLE
build request: add options to disable parts of the build

### ISSUE THREAD

**Issue by vtorri** [libjxl/libjxl#421]
what about : 

enable/disable static or shared lib
enable/disable doc
enable/disable binaries
enable/disable java

thank you

**Comment by deymo** [comment]
Manpages have `JPEGXL_ENABLE_MANPAGES`. The doxygen documentation is not part of the ALL target, so unless you build "doc" it won't be generated. Is there something else you would like to disabled from the doc?

I'm sending a patch to disable cjxl/djxl and to disable the Java bindings.

static/shared libs is a bit more complicated since they are not equivalent (we only expose the C API on the shared lib). We currently build both, but they are compiled only once. static/shared would require a lot more work to sort out, since if you request to "disable" the static libraries and only compile shared libs, cjxl/djxl still require to use the static libs.
